storage/remote/queue_manager.go

```diff
@@ -837,42 +839,75 @@ func (s *shards) sendSamples(ctx context.Context, samples []prompb.TimeSeries, b
 func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.TimeSeries, buf *[]byte) error {
 	backoff := s.qm.cfg.MinBackoff
 	req, highest, err := buildWriteRequest(samples, *buf)
+	reqSize := len(*buf)
+	sampleCount := len(samples)
 	*buf = req
 	if err != nil {
 		// Failing to build the write request is non-recoverable, since it will
 		// only error if marshaling the proto to bytes fails.
 		return err
 	}
 
+	try := 0
+
+	// This anonymous function is used to allow us to defer the finishing of our per-try span
+	// without causing a memory leak, and it has the nice effect of avoiding propagating most of
+	// the parameters for sendSamplesWithBackoff/3.
+	attemptStore := func() error {
+		span, ctx := opentracing.StartSpanFromContext(ctx, "Remote Send Batch")
+		defer span.Finish()
+
+		span.SetTag("samples", sampleCount)
+		span.SetTag("requestSize", reqSize)
+		span.SetTag("try", try)
+		span.SetTag("remote_name", s.qm.storeClient.Name())
+		span.SetTag("remote_url", s.qm.storeClient.Endpoint())
+
+		begin := time.Now()
+		err := s.qm.client().Store(ctx, *buf)
+		s.qm.metrics.sentBatchDuration.Observe(time.Since(begin).Seconds())
+
+		if err != nil {
+			span.LogKV("error", err)
+			ext.Error.Set(span, true)
+			level.Warn(s.qm.logger).Log("msg", "Failed to send batch, retrying", "err", err)
```

Comment:
This log line is still in the incorrect place. It should be with the `retriedSamplesTotal.Add()` metric call.